src/torchcodec/_core/DeviceInterface.cpp

```diff
@@ -12,8 +12,12 @@ namespace facebook::torchcodec {
 
 namespace {
 using DeviceInterfaceMap = std::map<torch::DeviceType, CreateDeviceInterfaceFn>;
-std::mutex g_interface_mutex;
-std::unique_ptr<DeviceInterfaceMap> g_interface_map;
+static std::mutex g_interface_mutex;
+
+DeviceInterfaceMap& getDeviceMap() {
```

Comment:
Makes sense. Thank you for the fix, @scotts.

Reply:
Curiously, this isn't the whole problem, but I suspect it's part of the problem. I also think we should make sure that we have a *use* of the boolean used to capture the result of registering the device. But, again, that hasn't solved all of the problems on my end, but I'm confident it's better than what we had before.

Reply:
@scotts, just to conclude the discussion. I see you've merged. Were the other changes needed on internal code base or the adjustments you've done actually fix the issues?